static/app/views/performance/browser/resources/resourceTable.tsx

```diff
@@ -25,7 +25,7 @@ type Row = {
   'resource.render_blocking_status': string;
   'span.description': string;
   'span.group': string;
-  'span.op': 'resource.script' | 'resource.img';
+  'span.op': `resource.${string}`;
```

Comment:
l: if we extract the type here we can make sure that `opNameMap` is typed with the correct key names.

there's also a fun TS hack you can do to get better autocomplete working here, but no need to add it.

```ts
type LiteralUnion<T extends string> = T | Omit<T, T>;

type SpanOp = `resource.${LiteralUnion<'script' | 'img' | 'css' | 'iframe'>}`
```

This will allow any string to be suffixed to `resource.X` but will fill in autocomplete when trying to look at possible resource values.

Reply:
hmm it worked for me in https://github.com/getsentry/sentry-javascript/blob/e507110d89fd8117bda0a83ad7f7d2ee88681615/packages/types/src/measurement.ts#L39-L41

give me a sec to play around with it.

Reply:
Sounds good, I do like that idea tho, we can probs figure another way to do it in the futre.